implementations/typescript/ockam/ockam_app/src/lib/TcpOutlets.svelte

```diff
@@ -0,0 +1,38 @@
+<script>
+    import { invoke } from '@tauri-apps/api/tauri'
+    import { listen, emit } from '@tauri-apps/api/event';
+
+    let tcp_outlets = invoke('tcp_outlet_list');
+
+    async function create() {
+        await invoke('tcp_outlet_create');
+    }
```

Comment:
I just started studying Svelte syntax last night, but is this a case where [reactive declaration](https://learn.svelte.dev/tutorial/reactive-declarations) or [reactive statements](https://learn.svelte.dev/tutorial/reactive-statements) might come in handy?

Reply:
How would you do it? To be honest, I struggled a little to make this part work and tried different approaches, so what I wrote might not be the optimal solution.

They way I had this working was: `create` button clicked -> call backend -> backend handles the call and emits an event -> frontend reacts to the event -> updates the `tcp_outlets` variable -> svelte reacts to the updated variable

The `tcp_outlets` variable is a promise that can be handled with an [svelte's await block](https://svelte.dev/examples/await-blocks).